docs/README.Ubuntu.md

```diff
@@ -0,0 +1,149 @@
+sudo add-apt-repository ppa:team-xbmc/xbmc-ppa-build-depends
+sudo apt update
+```
+
+Super-duper magic command to get the build dependencies:
+```
+sudo apt build-dep kodi
+```
+
+**WARNING:** Do not use `aptitude` for the `build-dep` command. It doesn't resolve everything properly.
+
+If at a later point you decide you do not want Kodi's PPAs on your system, removing them is as easy as:
+```
+sudo add-apt-repository -r ppa:team-xbmc/xbmc-nightly
+sudo add-apt-repository -r ppa:team-xbmc/xbmc-ppa-build-depends
+```
+
+**NOTE:** For developers and anyone else who builds frequently it is recommended to install `ccache` to expedite subsequent builds of Kodi.
+
+You can install it with:
+```
+sudo apt install ccache
+```
+
+**TIP:** If you have multiple computers at home, `distcc` will distribute build workloads of C and C++ code across several machines on a network. Team Kodi may not be willing to give support if problems arise using such a build configuration.
+
+You can install it with:
+```
+sudo apt install distcc
+```
+
+### 3.2. Get build dependencies manually
+If you get a `package not found` type of message with the below command, remove the offending package(s) from the install list and reissue the command. Take a note of the missing dependencies and, after a successful step completion, **[build the missing dependencies manually](#33-build-missing-dependencies)**.
+
+Install build dependencies manually:
+```
+sudo apt install debhelper autoconf automake autopoint gettext autotools-dev cmake curl default-jre doxygen gawk gcc gdc gperf libasound2-dev libass-dev libavahi-client-dev libavahi-common-dev libbluetooth-dev libbluray-dev libbz2-dev libcdio-dev libcec4-dev libp8-platform-dev libcrossguid-dev libcurl4-openssl-dev libcwiid-dev libdbus-1-dev libegl1-mesa-dev libenca-dev libflac-dev libfontconfig-dev libfmt3-dev libfreetype6-dev libfribidi-dev libgcrypt-dev libgif-dev libgles2-mesa-dev libgl1-mesa-dev libglu1-mesa-dev libgnutls28-dev libgpg-error-dev libiso9660-dev libjpeg-dev liblcms2-dev liblircclient-dev libltdl-dev liblzo2-dev libmicrohttpd-dev libmysqlclient-dev libnfs-dev libogg-dev libpcre3-dev libplist-dev libpng-dev libpulse-dev libshairplay-dev libsmbclient-dev libsqlite3-dev libssl-dev libtag1-dev libtiff5-dev libtinyxml-dev libtool libudev-dev libva-dev libvdpau-dev libvorbis-dev libxkbcommon-dev libxmu-dev libxrandr-dev libxslt1-dev libxt-dev lsb-release python-dev python-pil python-imaging rapidjson-dev swig unzip uuid-dev yasm zip zlib1g-dev
+```
+**WARNING:** Make sure you copy paste the entire line or you might receive an error or miss a few dependencies.
+
+Building for Wayland requires some extra packages:
+```
+sudo apt install libwayland-dev waylandpp-dev wayland-protocols
+```
+
+**WARNING:** Ubuntu 16.04 repositories don't have install candidates for `waylandpp-dev`and `wayland-protocols`. See **[build missing dependencies manually](#33-build-missing-dependencies)** section before you proceed.
```

Comment:
waylandpp-dev is not in any official ubuntu/debian repo, it's only available from our PPA